### DISCUSSION TITLE
LAB C - DSP Library: run makefile failed

### DISCUSSION THREAD

**Discussion by freud96** [bol-edu/HLS-SOC-Discussions#45] [General]
Hello, I am doing Lab-C, DSP vitis library. I have tried to use makefile to run sw/hw emulation, however I get this error below: 
   
34 | #pragma message("cl_version.h: CL_TARGET_OPENCL_VERSION is not defined. Defaulting to 220 (OpenCL 2.2)")
      |                                                                                                        ^
/usr/bin/ld: cannot find -lgmp
/usr/bin/ld: cannot find -lmpfr
/usr/bin/ld: cannot find -lIp_floating_point_v7_1_bitacc_cmodel

I think it's caused due to this line in the makefile: LDFLAGS += -pthread -L$(XILINX_XRT)/lib -L$(XILINX_HLS)/lnx64/tools/fpo_v7_1 -Wl,--as-needed -lOpenCL -lxrt_coreutil -lgmp -lmpfr -lIp_floating_point_v7_1_bitacc_cmodel 

Is there a way I can either those library or not using them at all?

**Comment by kevinjantw** [comment]
Hi,
Could you provide Lab-C DSP offical github link?

**Reply by freud96** [reply]
sorry for the late reply, "https://github.com/Xilinx/Vitis_Libraries/tree/main/dsp"  is the github link

**Comment by kevinjantw** [comment]
Hi @freud96,
The github link don't provide enough info.
Please you ref the two posts to reproduce your errors step-by-step.
https://github.com/bol-edu/HLS-Discussions/discussions/46
https://github.com/bol-edu/HLS-Discussions/discussions/43#discussioncomment-5681083
